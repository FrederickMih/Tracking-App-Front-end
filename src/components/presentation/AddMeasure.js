import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import axios from 'axios';
import Header from '../container/Header';
import '../../styles/AddMeasure.css';

const AddMeasure = () => {
  const history = useHistory();
  const [measurementId, setMeasurementId] = useState({
    id: 1,
  });

  const [measureValue, setMeasureValue] = useState({
    value: 0.01,
  });

  const measurements = useSelector((state) => state.measurements.measurements);

  const handleSelectChange = (e) => {
    setMeasurementId({
      id: e.target.value,
    });
  };

  const handleInputChange = (e) => {
    setMeasureValue({
      value: e.target.value,
    });
  };

  const handleClick = (e) => {
    e.preventDefault();
    const payload = {
      measurement: {
        id: measurementId.id,
      },
      measure: {
        value: measureValue.value,
      },
    };
    axios
      .post('https://guarded-sands-43543.herokuapp.com/measurements', payload)
      .then(() => {
        document.getElementById('measurement-input').value = '';
        history.push(`/progress/${measurementId.id}`);
      });
  };

  return (
    <div>
      <Header title="Add Measurement" />
      <div className="add-measure">
        <div className="container">
          <div className="select-wrapper">
            <select
              name="select-measurements"
              id="select-measurements"
              onChange={handleSelectChange}
            >
              {measurements.map((measurement) => (
                <option key={measurement.id} value={measurement.id}>
                  {measurement.name}
                </option>
              ))}
            </select>
            <input
              id="measurement-input"
              type="number"
              step="0.01"
              min="0"
              placeholder="0.01 cm"
              onChange={handleInputChange}
            />
          </div>
          <button
            className="AddMeasure__btn"
            type="submit"
            onClick={handleClick}
          >
            Add Measure
          </button>
        </div>
      </div>
    </div>
  );
};

export default AddMeasure;
