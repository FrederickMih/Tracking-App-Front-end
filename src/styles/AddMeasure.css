.add-measure,
.login-page {
  height: 82%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.container {
  width: 90%;
  background: white;
  padding: 2rem 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.container h1 {
  color: #575757;
  text-shadow: 0.2rem 0.2rem 0.5rem var(--main);
}

.select-wrapper {
  display: flex;
  flex-direction: row;
}

/* select, */

/* input, */
.AddMeasure__btn {
  font-size: 16px;
  letter-spacing: 2px;
  font-weight: 400 !important;
  background-color: teal;
  color: #fff;
  padding: 23px 50px;
  margin: auto;
  text-align: center;
  display: inline-block !important;
  text-decoration: none;
  border: 0;
  width: max-content;
  cursor: pointer;
  transition: all 0.3s 0s ease-in-out;

  /* margin: 4.5rem 4.5rem; */
}

.AddMeasure__btn:hover {
  transform: scale(1.04, 1);
  background-color: rgb(10, 209, 216);
  text-decoration: none;
}

/* select {
  padding: 0.25rem;
  width: 50%;
} */

.add-measure input {
  padding: 0.25rem;
  width: 50%;
}

select::-ms-expand {
  display: none;
}

select {
  width: 50%;
  display: inline-block;
  box-sizing: border-box;
  padding: 2rem;
  border: 1px solid #eee;
  font: inherit;
  line-height: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-repeat: no-repeat;
  background-image:
    linear-gradient(45deg, transparent 50%, currentColor 50%),
    linear-gradient(135deg, currentColor 50%, transparent 50%);
  background-position: right 15px top 1em, right 10px top 1em;
  background-size: 5px 5px, 5px 5px;
}

@media (min-width: 992px) {
  .AddMeasure__btn {
    width: 36%;
  }
}

/* Medium devices (tablets, 768px and up) */

/* @media (min-width: 767px) {
  .AddMeasure__btn {
    width: 62.5%;
  }
} */
