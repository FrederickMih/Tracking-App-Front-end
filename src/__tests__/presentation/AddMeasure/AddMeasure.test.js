import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import AddMeasure from '../../../components/presentation/AddMeasure';
import configureAppStore from '../../../Redux/configureStore';

const store = configureAppStore();

describe('AddMeasure', () => {
  test('should match the snapshot of AddMeasure component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <AddMeasure />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
