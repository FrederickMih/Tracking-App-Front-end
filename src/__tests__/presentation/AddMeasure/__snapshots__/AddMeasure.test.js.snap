// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMeasure should match the snapshot of AddMeasure component 1`] = `
Array [
  <div
    className="headline"
  >
    Add Measure
    <div
      className="bkbar"
    >
      <a
        href="/track"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
      </a>
    </div>
  </div>,
  <div
    className="add-measure"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="react-select css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className="react-select__control css-1s2u09g-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="react-select__value-container css-319lph-ValueContainer"
          >
            <div
              className="react-select__placeholder css-14el2xx-placeholder"
              id="react-select-2-placeholder"
            >
              <div
                className=""
              >
                Select your measurement item
              </div>
            </div>
            <div
              className="react-select__input-container css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-2-listbox"
                aria-describedby="react-select-2-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-2-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className="react-select__input"
                disabled={false}
                id="react-select-2-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="react-select__indicators css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className="react-select__indicator-separator css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className="react-select__indicator react-select__dropdown-indicator css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <input
        onChange={[Function]}
        placeholder="Data in cm"
        required={true}
        step="0.01"
        type="number"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>,
]
`;
